### DISCUSSION TITLE
Our wheel is not supported

### DISCUSSION THREAD

**Discussion by ravachol** [gotzl/hid-fanatecff#105] [General]
Hi, great project! 

Any chance you'll support Forza Motorsport CSR Wheel, seeing how the Elite version is supported?

**Comment by gotzl** [comment]
Hi,
isn't it already supported/working? At least I've listed it as `(experimental: 0EB7:0011 CSR Elite/Forza Motorsport Wheel Base)
` in the readme ...

**Comment by ravachol** [comment]
Ok, I couldn't get a game to detect it. But might be doing something wrong.

**Reply by gotzl** [reply]
Games should detect the wheel already w/o this driver. This driver only adds FFB support. Which game did you try?

**Reply by ravachol** [reply]
Ah, ok. I don't think there is a driver that works on linux with my wheel. F1 23.

**Reply by gotzl** [reply]
> Ah, ok. I don't think there is a driver that works on linux with my wheel. F1 23.

There is ;) The standard `hid-generic` driver should apply for this wheel. Maybe you can post the last 50 lines of `dmesg` after you've plugged in the wheel, then we can see. There is also `evtest` where the wheel should show up.

**Reply by ravachol** [reply]
[57049.162535] usb 3-14: new full-speed USB device number 10 using xhci_hcd
[57054.514929] usb 3-14: unable to read config index 0 descriptor/start: -71
[57054.514935] usb 3-14: can't read configurations, error -71
[57054.635905] usb 3-14: new full-speed USB device number 11 using xhci_hcd
[57059.839833] usb 3-14: unable to read config index 0 descriptor/start: -71
[57059.839838] usb 3-14: can't read configurations, error -71
[57059.839880] usb usb3-port14: attempt power cycle
[57060.242597] usb 3-14: new full-speed USB device number 12 using xhci_hcd
[57065.389927] usb 3-14: string descriptor 0 read error: -71
[57065.389934] usb 3-14: New USB device found, idVendor=0eb7, idProduct=0011, bcdDevice=12.22
[57065.389936] usb 3-14: New USB device strings: Mfr=1, Product=3, SerialNumber=0
[57065.391267] usb 3-14: can't set config #1, error -71

**Reply by ravachol** [reply]
It actually works, in a different game, project cars 2. No FfB though. I installed your driver and ran:
sudo udevadm control --reload-rules && sudo udevadm trigger

**Reply by gotzl** [reply]
Ok, so then check that you see s.t. like this in dmesg after connecting the wheel
```
input: Fanatec FANATEC Wheel as /devices/pci0000:00/0000:00:01.2/0000:01:00.0/0000:02:08.0/0000:0a:00.1/usb1/1-3/1-3:1.0/0003:0EB7:0020.000B/input/input28
fanatec 0003:0EB7:0020.000B: input,hiddev96,hidraw0: USB HID v1.11 Joystick [Fanatec FANATEC Wheel] on usb-0000:0a:00.1-3/input0
fanatec 0003:0EB7:0020.000B: Hires timer: period = 2 ms
```
for you obviously the names will be different, but you should also see the `fanatec` at the beginning of the tow last lines.
The try `fftest` and check if you get some movement there.

**Reply by ravachol** [reply]
[62177.195938] usb 3-14: new full-speed USB device number 27 using xhci_hcd
[62182.433122] usb 3-14: unable to read config index 0 descriptor/start: -71
[62182.433128] usb 3-14: can't read configurations, error -71
[62182.555972] usb 3-14: new full-speed USB device number 28 using xhci_hcd
[62182.698319] usb 3-14: string descriptor 0 read error: -32
[62182.698325] usb 3-14: New USB device found, idVendor=0eb7, idProduct=0011, bcdDevice=12.22
[62182.698327] usb 3-14: New USB device strings: Mfr=1, Product=3, SerialNumber=0
[62182.701042] input: HID 0eb7:0011 as /devices/pci0000:00/0000:00:14.0/usb3/3-14/3-14:1.0/0003:0EB7:0011.000A/input/input35
[62182.701376] hid-generic 0003:0EB7:0011.000A: input,hidraw5: USB HID v1.11 Joystick [HID 0eb7:0011] on usb-0000:00:14.0-14/input0

that's what I get. It doesn't say fanatec.

**Reply by gotzl** [reply]
This means that the driver isn't installed properly. Check if `modinfo hid-fanatec` works, if not try to install the driver again (`make install`) or load the driver directly with `sudo insmod hid-fanatec.ko`.

**Reply by ravachol** [reply]
sudo insmod hid-fanatec.ko
insmod: ERROR: could not insert module hid-fanatec.ko: Invalid module format

I'm on manjaro btw.

**Reply by ravachol** [reply]
[70404.220491] hid_fanatec: version magic '6.6.59-1-MANJARO SMP preempt mod_unload ' should be '6.6.60-1-MANJARO SMP preempt mod_unload '

**Reply by gotzl** [reply]
You've first installed the driver when you had `linux-6.6.59-1`, now you have `linux-6.6.60-1`. Currently, you've to re-compile/install the driver every time the kernel updates...

**Reply by ravachol** [reply]
ok cool, reinstalled.

[71636.196369] Buffer I/O error on dev sdb5, logical block 44, async page read
[71636.196377] ata2: EH complete
[71639.688881] ata2.00: exception Emask 0x0 SAct 0x2000 SErr 0x40000 action 0x0
[71639.688886] ata2.00: irq_stat 0x40000008
[71639.688887] ata2: SError: { CommWake }
[71639.688889] ata2.00: failed command: READ FPDMA QUEUED
[71639.688890] ata2.00: cmd 60/08:68:60:51:46/00:00:57:00:00/40 tag 13 ncq dma 4096 in
                        res 51/40:08:60:51:46/00:00:57:00:00/40 Emask 0x409 (media error) <F>
[71639.688894] ata2.00: status: { DRDY ERR }
[71639.688895] ata2.00: error: { UNC }
[71639.693120] ata2.00: configured for UDMA/133
[71639.693326] sd 1:0:0:0: [sdb] tag#13 FAILED Result: hostbyte=DID_OK driverbyte=DRIVER_OK cmd_age=3s
[71639.693330] sd 1:0:0:0: [sdb] tag#13 Sense Key : Medium Error [current] 
[71639.693332] sd 1:0:0:0: [sdb] tag#13 Add. Sense: Unrecovered read error - auto reallocate failed
[71639.693345] sd 1:0:0:0: [sdb] tag#13 CDB: Read(10) 28 00 57 46 51 60 00 00 08 00
[71639.693346] I/O error, dev sdb, sector 1464226144 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 2
[71639.693349] Buffer I/O error on dev sdb5, logical block 44, async page read
[71639.693358] ata2: EH complete
[71653.885307] usb 3-14: USB disconnect, device number 28
[71663.428688] usb 3-14: new full-speed USB device number 29 using xhci_hcd
[71668.782152] usb 3-14: unable to read config index 0 descriptor/all
[71668.782157] usb 3-14: can't read configurations, error -110
[71668.905323] usb 3-14: new full-speed USB device number 30 using xhci_hcd
[71674.115861] usb 3-14: unable to read config index 0 descriptor/start: -71
[71674.115865] usb 3-14: can't read configurations, error -71
[71674.115911] usb usb3-port14: attempt power cycle
[71674.525394] usb 3-14: new full-speed USB device number 31 using xhci_hcd
[71679.662689] usb 3-14: unable to read config index 0 descriptor/start: -71
[71679.662694] usb 3-14: can't read configurations, error -71
[71679.788772] usb 3-14: new full-speed USB device number 32 using xhci_hcd
[71684.998844] usb 3-14: unable to read config index 0 descriptor/all
[71684.998849] usb 3-14: can't read configurations, error -110
[71684.998890] usb usb3-port14: unable to enumerate USB device

**Reply by SynthetikzZ** [reply]
Did you recompiled the driver? Before sudo make install you do make

**Reply by ravachol** [reply]
yes, recompiled make clean make make install. then the other line.

now I rebooted and did it again:

[  147.550671] ata2.00: configured for UDMA/133
[  147.550907] sd 1:0:0:0: [sdb] tag#18 FAILED Result: hostbyte=DID_OK driverbyte=DRIVER_OK cmd_age=3s
[  147.550912] sd 1:0:0:0: [sdb] tag#18 Sense Key : Medium Error [current] 
[  147.550924] sd 1:0:0:0: [sdb] tag#18 Add. Sense: Unrecovered read error - auto reallocate failed
[  147.550926] sd 1:0:0:0: [sdb] tag#18 CDB: Read(10) 28 00 57 46 51 60 00 00 08 00
[  147.550927] I/O error, dev sdb, sector 1464226144 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 2
[  147.550930] Buffer I/O error on dev sdb5, logical block 44, async page read
[  147.550940] ata2: EH complete
[  151.036573] ata2.00: exception Emask 0x0 SAct 0x40 SErr 0x40000 action 0x0
[  151.036578] ata2.00: irq_stat 0x40000008
[  151.036579] ata2: SError: { CommWake }
[  151.036581] ata2.00: failed command: READ FPDMA QUEUED
[  151.036582] ata2.00: cmd 60/08:30:60:51:46/00:00:57:00:00/40 tag 6 ncq dma 4096 in
                        res 51/40:08:60:51:46/00:00:57:00:00/40 Emask 0x409 (media error) <F>
[  151.036586] ata2.00: status: { DRDY ERR }
[  151.036587] ata2.00: error: { UNC }
[  151.040526] ata2.00: configured for UDMA/133
[  151.040688] sd 1:0:0:0: [sdb] tag#6 FAILED Result: hostbyte=DID_OK driverbyte=DRIVER_OK cmd_age=3s
[  151.040691] sd 1:0:0:0: [sdb] tag#6 Sense Key : Medium Error [current] 
[  151.040693] sd 1:0:0:0: [sdb] tag#6 Add. Sense: Unrecovered read error - auto reallocate failed
[  151.040695] sd 1:0:0:0: [sdb] tag#6 CDB: Read(10) 28 00 57 46 51 60 00 00 08 00
[  151.040696] I/O error, dev sdb, sector 1464226144 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 2
[  151.040699] Buffer I/O error on dev sdb5, logical block 44, async page read
[  151.040707] ata2: EH complete
[  161.235651] usb 3-14: new full-speed USB device number 4 using xhci_hcd
[  166.399342] usb 3-14: unable to read config index 0 descriptor/all
[  166.399346] usb 3-14: can't read configurations, error -110
[  166.522554] usb 3-14: new full-speed USB device number 5 using xhci_hcd
[  171.732863] usb 3-14: unable to read config index 0 descriptor/all
[  171.732876] usb 3-14: can't read configurations, error -110
[  171.732997] usb usb3-port14: attempt power cycle
[  172.136173] usb 3-14: new full-speed USB device number 6 using xhci_hcd
[  172.155751] usb 3-14: string descriptor 0 read error: -32
[  172.155765] usb 3-14: New USB device found, idVendor=0eb7, idProduct=0011, bcdDevice=12.22
[  172.155773] usb 3-14: New USB device strings: Mfr=1, Product=3, SerialNumber=0
[  172.159741] input: HID 0eb7:0011 as /devices/pci0000:00/0000:00:14.0/usb3/3-14/3-14:1.0/0003:0EB7:0011.0006/input/input30
[  172.160339] hid-generic 0003:0EB7:0011.0006: input,hidraw5: USB HID v1.11 Joystick [HID 0eb7:0011] on usb-0000:00:14.0-14/input0
[  172.766446] input: HID 0eb7:0011 as /devices/pci0000:00/0000:00:14.0/usb3/3-14/3-14:1.0/0003:0EB7:0011.0006/input/input31
[  172.766771] fanatec 0003:0EB7:0011.0006: input,hidraw5: USB HID v1.11 Joystick [HID 0eb7:0011] on usb-0000:00:14.0-14/input0
[  172.766868] fanatec 0003:0EB7:0011.0006: Hires timer: period = 2 ms

**Reply by gotzl** [reply]
Just a quick note, all these usb errors tell me that there's s.t. wrong with the usb connection, not sure if this is the device, the cable or your mainboard, but its not related to the driver/linux.

**Reply by ravachol** [reply]
ok at least it says fanatec now. FFB didn't work in F1 22. F1 23 doesn't work with the wheel at all it seems.

**Reply by ravachol** [reply]
Thanks for the help.

**Reply by ravachol** [reply]
Maybe add a note in your readme that you need to recompile for every kernel.

**Reply by gotzl** [reply]
> ok at least it says fanatec now. FFB didn't work in F1 22. F1 23 doesn't work with the wheel at all it seems.

Can you test with `fftest` first? This is important to know if FFB is working on linux level.

**Reply by ravachol** [reply]
I had to update the system and recompile again. 😅 fftest does not work at all.

**Reply by gotzl** [reply]
What do you mean with ` fftest does not work at all.`?

**Reply by ravachol** [reply]
there is no force feedback on the wheel when i try the different settings. the program runs though.

**Reply by gotzl** [reply]
Ah, ok. And this time you've confirmed that dmesg shows `fanatec` as the driver?

**Reply by ravachol** [reply]
it does, but pretty high up.

If I disconnect and reconnect the wheel it doesn't say fanatec again.

[ 1943.706739] ata2: EH complete
[ 1947.189240] ata2.00: exception Emask 0x0 SAct 0x2000 SErr 0x40000 action 0x0
[ 1947.189258] ata2.00: irq_stat 0x40000008
[ 1947.189259] ata2: SError: { CommWake }
[ 1947.189261] ata2.00: failed command: READ FPDMA QUEUED
[ 1947.189262] ata2.00: cmd 60/08:68:60:51:46/00:00:57:00:00/40 tag 13 ncq dma 4096 in
                        res 51/40:08:60:51:46/00:00:57:00:00/40 Emask 0x409 (media error) <F>
[ 1947.189266] ata2.00: status: { DRDY ERR }
[ 1947.189267] ata2.00: error: { UNC }
[ 1947.193219] ata2.00: configured for UDMA/133
[ 1947.193468] sd 1:0:0:0: [sdb] tag#13 FAILED Result: hostbyte=DID_OK driverbyte=DRIVER_OK cmd_age=3s
[ 1947.193480] sd 1:0:0:0: [sdb] tag#13 Sense Key : Medium Error [current] 
[ 1947.193486] sd 1:0:0:0: [sdb] tag#13 Add. Sense: Unrecovered read error - auto reallocate failed
[ 1947.193492] sd 1:0:0:0: [sdb] tag#13 CDB: Read(10) 28 00 57 46 51 60 00 00 08 00
[ 1947.193496] I/O error, dev sdb, sector 1464226144 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 2
[ 1947.193507] Buffer I/O error on dev sdb5, logical block 44, async page read
[ 1947.193532] ata2: EH complete
[ 1949.512153] usb 3-14: new full-speed USB device number 8 using xhci_hcd
[ 1954.722589] usb 3-14: string descriptor 0 read error: -71
[ 1954.722607] usb 3-14: New USB device found, idVendor=0eb7, idProduct=0011, bcdDevice=12.22
[ 1954.722615] usb 3-14: New USB device strings: Mfr=1, Product=3, SerialNumber=0
[ 1954.724765] usb 3-14: can't set config #1, error -71

**Reply by gotzl** [reply]
The things high-up are not relevant, only the most recent stuff after `usb 3-14: New USB device found, idVendor=0eb7, idProduct=0011, bcdDevice=12.22`. It's still puzzling for me why you have so many USB errors, this is not normal/good. I think you've to first try to get a stable USB connection before we can proceed.

**Reply by ravachol** [reply]
I've been running this system for years..and have tried every little thing. so it's pretty messed up. But I don't know about usb or how I would go about fixing it.

**Reply by ravachol** [reply]
I'm glad I have a working wheel now though, hats off.

**Reply by gotzl** [reply]
Ok, for me its hard to understand what's going on.. Maybe try it like this, run `journalctl -xkf` in on terminal and observe it all time. Then plug-in the wheel and observe what's happening. There should be this message `[ 1954.722607] usb 3-14: New USB device found, idVendor=0eb7, idProduct=0011, bcdDevice=12.22` and a few lines later followed `[ 172.766771] fanatec 0003:0EB7:0011.0006: ....`. Then use `evtest` to see, which event device file is assigned to the wheel, and use `fftest` with this event device file. This should give you some forces on the wheel ;)
When doing `evtest` and `ffbtest`, there shoul be no more lines in the journal about `[ 1954.722589] usb 3-14:` ...

**Reply by ravachol** [reply]
nov 17 22:32:12 x kernel: usb 3-14: new full-speed USB device number 5 using xhci_hcd
nov 17 22:32:18 x kernel: usb 3-14: unable to read config index 0 descriptor/all
nov 17 22:32:18 x kernel: usb 3-14: can't read configurations, error -110
nov 17 22:32:18 x kernel: usb 3-14: new full-speed USB device number 6 using xhci_hcd
nov 17 22:32:23 x kernel: usb 3-14: unable to read config index 0 descriptor/all
nov 17 22:32:23 x kernel: usb 3-14: can't read configurations, error -110
nov 17 22:32:23 x kernel: usb usb3-port14: attempt power cycle
nov 17 22:32:23 x kernel: usb 3-14: new full-speed USB device number 7 using xhci_hcd
nov 17 22:32:29 x kernel: usb 3-14: unable to read config index 0 descriptor/all
nov 17 22:32:29 x kernel: usb 3-14: can't read configurations, error -110
nov 17 22:32:29 x kernel: usb 3-14: new full-speed USB device number 8 using xhci_hcd
nov 17 22:32:34 x kernel: usb 3-14: unable to read config index 0 descriptor/all
nov 17 22:32:34 x kernel: usb 3-14: can't read configurations, error -110
nov 17 22:32:34 x kernel: usb usb3-port14: unable to enumerate USB device

**Reply by gotzl** [reply]
Well, all these messages say, that the usb device is not connecting at all, on the 'physical' layer. Ie, if you do `lsusb`, the device is not listed...

**Reply by ravachol** [reply]
it is not.

**Reply by gotzl** [reply]
Jea, this I cannot fix ;) Do you use some kind of hub or USB extension? Maybe try another cable, or another USB port ....

**Reply by gotzl** [reply]
> yet I could use it in F1 22

Seems that the connection worked for some time, but now it seems the USB connection to not work anymore...

**Reply by ravachol** [reply]
unfortunately, that seems to be the case.

**Reply by ravachol** [reply]
lsusb Endor AG shows up.

**Reply by gotzl** [reply]
And in `evtest`?

**Reply by ravachol** [reply]
633.943046] fanatec 0003:0EB7:0011.0006: input,hidraw5: USB HID v1.11 Joystick [HID 0eb7:0011] on usb-0000:00:14.0-5/input0
[  633.943142] fanatec 0003:0EB7:0011.0006: Hires timer: period = 2 ms

And got it to work in F1 22.

Had to unplug the usb cable, then plug it and restart it.

**Reply by ravachol** [reply]
> And in `evtest`?

It says /dev/input/event30:     HID 0eb7:0011

**Reply by gotzl** [reply]
Hmm.. ok, but no FFB with `fftest /dev/input/event30`?

**Reply by ravachol** [reply]
nope

**Reply by gotzl** [reply]
Ok, what about `lsmod | grep hid` and `ls /sys/module/hid_fanatec/drivers/hid\:fanatec/`?

**Comment by ravachol** [comment]
lsmod | grep hid
hid_fanatec            53248  0
mac_hid                12288  0
usbhid                 86016  0

ls /sys/module/hid_fanatec/drivers/hid\:fanatec/?
ls: cannot access '/sys/module/hid_fanatec/drivers/hid:fanatec/?': No such file or directory

EDIT: sorry it says:
ls /sys/module/hid_fanatec/drivers/hid:fanatec/
0003:0EB7:0011.0006  bind  module  new_id  uevent  unbind

**Reply by gotzl** [reply]
Ok, so this all looks good to me... Not sure why `fftest` doesn't work for you then. I'll think about it, have to go to bed now...

**Reply by ravachol** [reply]
ok, if you need any help with testing...I'll help.
